fw/http.c

```diff
@@ -1307,17 +1489,133 @@ tfw_http_nip_req_resched_err(TfwSrvConn *srv_conn, TfwHttpReq *req,
 
 /* Common interface for sending error responses. */
 void
-tfw_http_send_resp(TfwHttpReq *req, int status, const char *reason)
+tfw_http_send_err_resp(TfwHttpReq *req, int status, const char *reason)
 {
 	if (!(tfw_blk_flags & TFW_BLK_ERR_NOLOG)) {
 		T_WARN_ADDR_STATUS(reason, &req->conn->peer->addr,
 				   TFW_NO_PORT, status);
 	}
 
 	if (TFW_MSG_H2(req))
-		tfw_h2_send_resp(req, status, 0);
+		tfw_h2_send_err_resp(req, status, 0);
 	else
-		tfw_h1_send_resp(req, status);
+		tfw_h1_send_err_resp(req, status);
+}
+
+static void
+tfw_http_send_resp(TfwHttpReq *req, TfwStr *msg, int status)
+{
+	if (TFW_MSG_H2(req)) {
+		tfw_h2_send_resp(req, msg, status, 0);
+	} else {
+		TfwCliConn *cli_conn = (TfwCliConn *)req->conn;
+
+		WARN_ONCE(!list_empty_careful(&req->msg.seq_list),
+			  "Request is already in seq_queue\n");
+		tfw_stream_unlink_msg(req->stream);
+		spin_lock(&cli_conn->seq_qlock);
+		list_add_tail(&req->msg.seq_list, &cli_conn->seq_queue);
+		spin_unlock(&cli_conn->seq_qlock);
+
+		tfw_h1_send_resp(req, msg, status);
+	}
+}
```

Comment:
Shouldn't the functions be `tfw_http_send_redir()`, `tfw_h1_send_redir()` and `tfw_h2_send_redir()`? They send redirects only, no any HTTP response